pkg/network/manualpeering/manualpeering.go

```diff
@@ -68,52 +29,38 @@ type KnownPeer struct {
 type Manager struct {
 	p2pm              *p2p.Manager
 	log               *logger.Logger
-	local             *peer.Local
 	startOnce         sync.Once
 	isStarted         atomic.Bool
 	stopOnce          sync.Once
 	stopMutex         syncutils.RWMutex
 	isStopped         bool
```

Comment:
Without an atomic bool it could happen that you still read false when it was actually true, for something read so infrequently and without consequences if you process just another loop I thought it was not really necessary. For the `isStarted` this reasoning could also apply, but there you might actually panic during initialization, as you are trying to process something on a component that should be started but you read it is not. I can change it, but I believe it is unnecessary.